server/src/test/java/org/elasticsearch/action/search/CanMatchPreFilterSearchPhaseTests.java

```diff
@@ -213,35 +209,13 @@ public void sendCanMatch(
                 timeProvider,
                 0,
                 null,
-                (iter) -> new InitialSearchPhase<SearchPhaseResult>("test", searchRequest,
-                        iter, logger, randomIntBetween(1, 32), executor) {
+                iter -> new SearchPhase("next") {
                     @Override
-                    void onPhaseDone() {
+                    public void run() {
                         latch.countDown();
                     }
-
```

Comment:
I am curious to hear if I am over-simplifying things here. This is how the next phase used to be before #27069 . I wonder if it's necessary for the next phase to be an AbstractSearchAsyncAction subclass at this point, which is not so easy to recreate.